pydicom/dataset.py

```diff
@@ -201,7 +205,12 @@ def _dict_equal(a, b, exclude=None):
             )
 
 
-class Dataset(dict):
+_Dataset = TypeVar("_Dataset", bound="Dataset")
+
+
+class Dataset(
+    Iterable[DataElement], Dict[BaseTag, Union[DataElement, RawDataElement]]
```

Comment:
Broken in Python 3.6 😭, it fixed so many mypy warnings since our dict iterable is backwards (values instead of keys)